### PR TITLE
[BugFix] Fix OlapTableSink close accelerate release resource

### LINKED ISSUE #8889: [Bug] Not cancel after try close failed
```
Status OlapTableSink::close(RuntimeState* state, Status close_status) {
    if (close_status.ok()) {
        do {
            RETURN_IF_ERROR(try_close(state)); 
            SleepFor(MonoDelta::FromMilliseconds(5));
        } while (!is_close_done());
    }
    return close_wait(state, close_status);
}
```

not cancel when try close failed

### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
2. INSERT INTO '....'
3. SELECT  '....'

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #8889

### CODE REVIEW

**Review by trueeyu — APPROVED** [review]

File: be/src/exec/tablet_sink.cpp [hunk 1]
```diff
@@ -1001,7 +1001,8 @@ bool OlapTableSink::is_close_done() {
 Status OlapTableSink::close(RuntimeState* state, Status close_status) {
     if (close_status.ok()) {
         do {
-            RETURN_IF_ERROR(try_close(state));
+            close_status = try_close(state);
+            if (!close_status.ok()) break;
```
Comment (nshangyiming) [inline be/src/exec/tablet_sink.cpp:1005]:
```suggestion
            if (!try_close(state).ok()) {
                break;
            }
```

Comment (meegoo) [reply be/src/exec/tablet_sink.cpp:1005]: close_status will be the parameter pass to close_wait()

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by meegoo** [comment]
@mergify backport branch-2.3

**Pull request merged**